src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,29 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can`t be null!");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must contain at least 6 symbols!");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login already exists!");
+        }
+
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null!");
+        }
+
+        if (user.getPassword().length() < 6) {
```

Comment:
Consider checking if the password is null before checking its length to avoid a NullPointerException. Although you have a null check on line 24, it's better to keep checks consistent.